R/methods-Module.R

```diff
@@ -10,127 +9,183 @@
 #' }
 #' @param tree whether to use tree as latent space. If TRUE, object should have
 #' a tree slot.
-#' @param number_top_genes hotspot argument for number of genes to consider
+#' @param number_top_genes Hotspot argument for number of genes to consider
 #' @param num_umi optional dataframe containing umi counts in first column for
 #'  barcodes
-#' @param min_gene_threshold minimum number of genes in hotspot module
+#' @param min_gene_threshold minimum number of genes in Hotspot module
 #' @param n_neighbors number of neighbors to consider in latent space
 #' @param autocorrelation_fdr threshold for significance for genes autocorr
 #' @param clustering_fdr threshold for significance for clustering modules
+#' @param nn_precomp precomputed neighbors, barcodes x neighbors
+#' @param wt_precomp precomputed weights, barcodes x neighbors
 #' 
 #' Populates the modData, HotspotModuleScores, ModuleSignatureEnrichment
 #' and HotspotObject slots of object, as well as recalculates signature scores
 #' for new modules.
-calcHotspotModules <- function(object, model="normal", tree=FALSE, 
-                               number_top_genes=1000,num_umi=NULL, 
+#' @return the modified Vision object
+#' 
+#' @export
+hsAnalyze <- function(object, model="normal", tree=FALSE, 
```

Comment:
Let's call this something more informative like `runHotspot` or something

Reply:
I went with hsAnalyze since it matches the VISION pattern of main analysis function just being called analyze

Reply:
I was more saying that using the name `hsAnalyze` is not clear because it's not clear _a priori_ that `hs` means `Hotspot`. Since `hotspotAnalyze` sounds a bit burdensome I suggested `runHotspot`.

It doesn't really matter to me what you name it as long as you don't use the abbreviation `hs.`

Reply:
Changed!